lib/ical/helpers.js

```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  * Portions Copyright (C) Philipp Kewisch, 2011-2015 */
 
-
+var ICAL;
 /* istanbul ignore next */
 /* jshint ignore:start */
 if (typeof module === 'object') {
```

Comment:
I'm thinking something like this would be a good start:

```javascript
/* istanbul ignore next */
/* jshint ignore:start */
if (typeof module === 'object') {
  // CommonJS, where exports may be different each time.
  var ICAL = module.exports;
} else if (typeof this !== "undefined" && typeof ICAL !== 'object') {/* istanbul ignore next */
  /** @ignore */
  this.ICAL = {};
} else if (typeof ICAL !== 'object') {
  var ICAL = {};
}
/* jshint ignore:end */
```

What I'm still not sure about is how to make things work in native es6 modules. To actually export `ICAL`, we need `export default ICAL`, but that again doesn't work when it is loaded from a non-module. I don't want to use eval, I guess the only other option is to transpile?

Reply:
Also now I look - anyway that `var` in the `if` is effectively lifted to the scope of the file - having `var ICAL` in two parts of the `if` causes a warning of duplicated definition.

Reply:
I thought this would not be the case given it is in two different branches, only one can ever be called. Happy to move this up though. I believe the `this.ICAL` case was added for use in `Components.utils.import()` which I am going to be bold and drop now. Taking #450 into account as well, what do you think about this?

```javascript

/* jshint ignore:start */
var ICAL;
/* istanbul ignore next */
if (typeof module === 'object') {
  // CommonJS, where exports may be different each time.
  ICAL = module.exports;
} else if (typeof HTMLScriptElement !== 'undefined' && 'noModule' in HTMLScriptElement.prototype) {
  // Until we use ES6 exports, using <script type="module"> we define ICAL on the window global.
  window.ICAL = ICAL = {};
} else if (typeof ICAL !== 'object') {
  ICAL = {};
}
/* jshint ignore:end */
```

Reply:
The check for nomodule - basically whats the point? when we have a top level `var` - in a <script src=ical> that var will be exactly the same result as assigning to window.ICAl

Reply:
This was to cope with <script type=module> as in #450 . The global scope is not the window iirc, and until we officially support modules it would help make ICAL global.